doc/SAI-Proposal-ACL-1.md

```diff
@@ -0,0 +1,89 @@
+SAI access control lists (ACL) enhancements for SAI 1.0 release
+-------------------------------------------------------------------------------
+
+ Title       | SAI ACL Model - Enhancements
+-------------|----------------------
+ Authors     | Cavium Inc.
+ Status      | In review
+ Type        | Standards track
+ Created     | 08/09/2016
+ SAI-Version | 1.0.0
+
+-------------------------------------------------------------------------------
+
+## Overview ##
+SAI Access Control Lists (ACL) object implements ACL management functions. In SAI 0.9.1 through 0.9.5 versions, SAI ACL contained three types of objects, ACL table, ACL entry and ACL counter. The ACL table contains a number of ACL entries. Each ACL table defines a set of unique matching fields for all its ACL entries. A packet can match rules in different ACL tables and take non-conflicting actions from all the matched rules. However, within an ACL table, if a packet matches multiple rules, only the actions from the rule of highest priority are executed. ACL counters can also be created and attached to an ACL entry in order to counter the number of packets or bytes that match the ACL entry. The initial version of ACL table object has several ambiguities and limitations. We propose the following enhancements to address some of those and propose a generic and simple ACL model for operators:
```

Comment:
The idea to enable programming any ACL Group or ACL Table be applied to filter traffic INGRESSING or EGRESSING the switch (all ports). It would be filtering all traffic. The real reason to provide such a configuration is to provide backward compatibility to the Pre SAI1.0 ACL model. (Which was assuming such a behavior).